docs/router/src/example/building-a-nest.md

```diff
@@ -0,0 +1,263 @@
+an HTML anchor element but that would cause our page to reload unnecessarily.
+Instead we want to use the [`Link`] component provided by Dioxus Router.
+
+The [`Link`] is similar to a regular `a` tag. It takes a target (for now a path,
+more on other targets later) and an element. Let's add our links
+
+```rust,no_run
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+# use dioxus::prelude::*;
+#
+#[allow(non_snake_case)]
+fn NavBar(cx: Scope) -> Element {
+    cx.render(rsx! {
+        nav {
+            ul {
+                // new stuff starts here
+                li {
+                    Link {
+                        target: NtPath(String::from("")),
+                        "Home"
+                    }
+                }
+                li {
+                    Link {
+                        target: "/blog".into(), //short form
+                        "Blog"
+                    }
+                }
+                // new stuff ends here
+            }
+        }
+    })
+}
+```
+
+> Using this method, the [`Link`] component only works for links within our
+> application. To learn more about navigation targets see
+> [here](./navigation-targets.md).
+
+And finally, we add the navbar component in our app component:
+```rust,no_run
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+# use dioxus::prelude::*;
+# fn Home(cx: Scope) -> Element { unimplemented!() }
+# fn NavBar(cx: Scope) -> Element { unimplemented!() }
+# fn PageNotFound(cx: Scope) -> Element { unimplemented!() }
+#
+#[allow(non_snake_case)]
+fn App(cx: Scope) -> Element {
+    let routes = use_segment(&cx, || {
+        Segment::new().index(RcComponent(Home))
+    });
+
+    cx.render(rsx! {
+        Router {
+            routes: routes.clone(),
+            fallback: RcComponent(PageNotFound),
+            NavBar { } // this is new
+            Outlet { }
+        }
+    })
+}
+```
+Now you should see a list of links near the top of your page. Click on one and
+you should seamlessly travel between pages.
+
+### Active Link Styling
+You might want to style links differently, when their page is currently open.
+To achieve this, we can tell the router to give the internal `a` tag a class in
+that case.
+
+```rust,no_run
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+# use dioxus::prelude::*;
+# fn Home(cx: Scope) -> Element { unimplemented!() }
+# fn NavBar(cx: Scope) -> Element { unimplemented!() }
+# fn PageNotFound(cx: Scope) -> Element { unimplemented!() }
+#
+#[allow(non_snake_case)]
+fn App(cx: Scope) -> Element {
+    let routes = use_segment(&cx, || {
+        Segment::new().index(RcComponent(Home))
+    });
+
+    cx.render(rsx! {
+        Router {
+            routes: routes.clone(),
+            fallback: RcComponent(PageNotFound),
+            active_class: "active", // this is new
```

Comment:
Yeah, I'll remove it.

Reply:
I thought about it some more and now I'm not sure if removing this is the best thing to do. Leaving it might remove quite a bit of boiler plate in apps. For example, if you have a `nav` with a few dozen links, you might want all of them to have a class when they are active.

Reply:
Couldn't the user code just create a StyledLink component that wraps Link and provides that class (and any other presentational logic, like wrappers or whatever)?

The need for consistent styling is not unique to Router Links. Creating components to reuse the same styling is a common pattern.

Reply:
Yeah, you're right. I'll remove it then

Reply:
And it's gone